source/retryable-writes/retryable-writes.rst

```diff
@@ -474,8 +475,14 @@ The above rules are implemented in the following pseudo-code:
          * For exceptions that originate from the driver (e.g. no socket available
          * from the connection pool), we should raise the previous error if there
          * was one.
+         *
+         * In addition, if the previousError is null (i.e. this is the first
+         * attempt), we should persist the previousError as the originalError,
+         * in case we need to raise it later (e.g. if all errors are labeled
+         * "NoWritesPerformed").
          */
-        if (originalError is not DriverException && ! originalError.hasErrorLabel("NoWritesPerformed")) {
+        if (originalError is not DriverException && ! originalError.hasErrorLabel("NoWritesPerformed") ||
```

Comment:
I realize this predates this PR, but I'm finding it quite difficult to make sense of the current pseudo-code here. When I first wrote this for the two-attempt algorithm, the exceptions names were `originalError` and `secondError` (see: [this blame](https://github.com/mongodb/specifications/blame/6f21a767da943664d67bd967cb8ae97a35e40bba/source/retryable-writes/retryable-writes.rst#L450)).

It looks like things started to go downhill with changes for PoolCleared (see: [blame](https://github.com/mongodb/specifications/blame/88c583457c2481871ca6ab22be117a4da9d97135/source/retryable-writes/retryable-writes.rst#L461)), which introduced a syntax error (variable `e` was never defined). That stuck around in the first CSOT change (see: [blame](https://github.com/mongodb/specifications/blame/008d2f5aab41dabced03e0860bab614c0f15ded5/source/retryable-writes/retryable-writes.rst#L456)) until it was eventually removed in [343ff9a](https://github.com/mongodb/specifications/commit/343ff9a3864e7141c5a056fbf19e71ca57e65740#diff-01c94ffec48124f66d321265e57d6c892b1355813cf2bce099d0345ff222eabeR453) last month.

However, the variable name `originalError` stuck around. I think this should be renamed to `currentError` for clarity. That way, `currentError` is always the most recent exception and `previousError` gets set by this conditional.